qgis_plugin_gpq_downloader.py

```diff
@@ -661,36 +662,54 @@ def run(self, default_source=None):
         # Connect validation complete signal to handle the result
         dialog.validation_complete.connect(
             lambda success, message, results: self.handle_validation_complete(
-                success, message, results, dialog.get_url(), self.iface.mapCanvas().extent()
+                success, message, results, dialog.get_url(), self.iface.mapCanvas().extent(), dialog
             )
         )
         
         if dialog.exec_() != QDialog.Accepted:
             return
 
-    def handle_validation_complete(self, success, message, validation_results, dataset_url, extent):
-        """Handle validation completion and start download if successful"""
+    def handle_validation_complete(self, success, message, validation_results, url, extent, dialog):
+        """Handle validation completion and start download if successful."""
         if success:
-            # Generate default filename
-            from datetime import datetime
-            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
-            default_filename = f"geoparquet_download_{timestamp}.parquet"
-            default_save_path = str(self.download_dir / default_filename)
+            # Get current date for filename
+            current_date = datetime.datetime.now().strftime('%Y%m%d')
             
+            # Generate the default filename based on dialog selection
+            if dialog.overture_radio.isChecked():
+                theme = dialog.overture_combo.currentText().lower()
+                if theme == "base":
+                    subtype = dialog.base_subtype_combo.currentText()
+                    filename = f"overture_base_{subtype}_{current_date}.parquet"
+                else:
+                    filename = f"overture_{theme}_{current_date}.parquet"
+            
+            elif dialog.sourcecoop_radio.isChecked():
+                selection = dialog.sourcecoop_combo.currentText()
+                # Convert display name to safe filename format
+                safe_name = selection.lower().replace(" ", "_").replace("/", "_")
+                filename = f"sourcecoop_{safe_name}_{current_date}.parquet"
+            
+            else:  # custom URL
+                filename = f"custom_download_{current_date}.parquet"
```

Comment:
I think this works for the first iteration, but I do think we should add a ticket to try to get the file name of the remote url used. At least if it's a .parquet file, which is most of what I do with 'custom' (as opposed to the glob (*) urls - it'd be hard to automatically get a file name from those).

Reply:
Ah, I hadn't really taken the custom urls into consideration, was looking more at what comes out via Overture/SC

Lets pick that up in new ticket as suggested.